src/coreclr/jit/lclvars.cpp

```diff
@@ -1889,31 +1889,35 @@ void Compiler::lvaClassifyParameterABI()
 }
 
 //--------------------------------------------------------------------------------------------
-// lvaHaveSwiftStructStackParamsToReassemble:
-//   Check if this compilation has any Swift parameters that are passed on the
-//   stack and that need to be reassembled on the local stack frame.
+// lvaHasAnyStackParamToReassemble:
+//   Check if this compilation has any parameters split that need to be reassembled on the local stack frame.
 //
 // Return value:
 //   True if so.
 //
-bool Compiler::lvaHasAnySwiftStackParamToReassemble()
+bool Compiler::lvaHasAnyStackParamToReassemble()
 {
+#if defined(SWIFT_SUPPORT) || defined(TARGET_RISCV64)
 #ifdef SWIFT_SUPPORT
     if (info.compCallConv != CorInfoCallConvExtension::Swift)
     {
         return false;
     }
+#endif
 
     for (unsigned lclNum = 0; lclNum < info.compArgsCount; lclNum++)
     {
         const ABIPassingInformation& abiInfo = lvaGetParameterABIInfo(lclNum);
+#ifdef SWIFT_SUPPORT
         if (abiInfo.HasAnyStackSegment() && !abiInfo.HasExactlyOneStackSegment())
+#else // TARGET_RISCV64
+        if (abiInfo.IsSplitAcrossRegistersAndStack())
+#endif
         {
             return true;
         }
     }
-#endif
-
+#endif // defined(SWIFT_SUPPORT) || defined(TARGET_RISCV64)
```

Comment:
I would split this into a loop for the Swift case and a loop for the RISCV case. They seem to be separate cases and RISCV would want to have both loops if they are going to support the Swift ABI in the future.

Reply:
On second thought I think you need to call this function in the prolog for RISCV64. Otherwise you can end up with garbage GC pointer in the split local's home after the prolog, which is not ok. (For Swift we don't have this problem because we won't have any GC pointers in parameters.)